### ISSUE TITLE
Waves.Exchange fetch_trades() tags all returned trades as 'sell'

### ISSUE THREAD

**Issue by JovianMoon** [ccxt/ccxt#15853]
- OS: Linux Mint
- Programming Language version: Python 3.10
- CCXT version: 2.2.36

```
exchange = ccxt.wavesexchange()

total_trade_count = 0      
buy_trade_count = 0
sell_trade_count = 0  
maker_trade_count = 0
taker_trade_count = 0

trades = exchange.fetch_trades('ETH/BTC', limit=100)
for trade in trades:
    total_trade_count += 1
    if trade['side'] == 'buy':
        buy_trade_count += 1
    if trade['side'] == 'sell':
        sell_trade_count += 1
    if trade['takerOrMaker'] == 'maker':
        maker_trade_count += 1
    if trade['takerOrMaker'] == 'taker':
        taker_trade_count += 1        
    
print(trades[0])
print('total trades:', total_trade_count)
print('buy trades:  ', buy_trade_count)
print('sell trades: ', sell_trade_count)   
print('maker trades:', maker_trade_count)
print('taker trades:', taker_trade_count)
print()
```

```
{'info': {'__type': 'transaction', 'data': {'id': 'Fnc9rbxg87Uik6dSZeLnDzoYfY9C7g1mRjBxcwwniTrT', 'timestamp': '2022-11-26T03:17:34.263Z', 'height': '3398988', 'type': '7', 'version': '3', 'proofs': ['2xZzjHY5jKs7xSx6a6zx8PsYHX7Uf7Zr6fT5bT4eqqyK3ong9XFtp4Nnwojw3oraNUwGoYscsKFZofWXiKKij6eL'], 'fee': '0.003', 'applicationStatus': 'succeeded', 'sender': '3PEjHv3JGjcWNpYEEkif2w8NXV4kbhnoGgu', 'senderPublicKey': '9cpfKN9suPNvfeUNphzxXMjcnn974eme8ZhWUjaktzU5', 'buyMatcherFee': '0.11868737', 'sellMatcherFee': '0.11888352', 'price': '0.0729996', 'amount': '0.06195336', 'order1': {'id': 'HVhALdt4VopYfzpuszHz2FyfxH2uDv6odmfxGpDxYBbG', 'version': '3', 'senderPublicKey': '2c5vbaE3bwbZJHGHRYf8wALD16Q8du5vnwFScDHxyLXU', 'matcherPublicKey': '9cpfKN9suPNvfeUNphzxXMjcnn974eme8ZhWUjaktzU5', 'assetPair': {'amountAsset': '474jTeYx2r2Va35794tCScAXWJG9hU2HcgxzMowaZUnu', 'priceAsset': '8LQW8f7P5d5PZM7GtZEBgaqRPGSzS3DfPuiXrURJ4AJS'}, 'orderType': 'buy', 'price': '0.0729996', 'sender': '3PQFNSthBgj8A2zRSrJXKviuwtqxhWUp2WR', 'amount': '0.32', 'timestamp': '2022-11-26T03:17:10.850Z', 'expiration': '2022-11-27T03:17:10.850Z', 'matcherFee': '0.61309655', 'signature': '45zgQr1hsyqxAxmdcqvaPrwRmHzu6FK7T3LVZRzvtTetqUj7sw4pGsVaJYUuRe9kCEdHTbHdhwbxTYokVAZfaJ73', 'matcherFeeAssetId': 'Atqv59EYzjFGuitKVnMRk6H8FukjoV3ktPorbEys25on'}, 'order2': {'id': '8BcAp5PDVAZup99Sdo4it8hTm93HjCeTrTkSCkQ4sAYq', 'version': '3', 'senderPublicKey': 'FpZAEi3Ra4LuSmUeBV6hsosVHqhq8667LduGWZDeyAbV', 'matcherPublicKey': '9cpfKN9suPNvfeUNphzxXMjcnn974eme8ZhWUjaktzU5', 'assetPair': {'amountAsset': '474jTeYx2r2Va35794tCScAXWJG9hU2HcgxzMowaZUnu', 'priceAsset': '8LQW8f7P5d5PZM7GtZEBgaqRPGSzS3DfPuiXrURJ4AJS'}, 'orderType': 'sell', 'price': '0.0729996', 'sender': '3P5wPrvSgxks5NzTPGDJXokT9PKQBSAp8En', 'amount': '0.06195349', 'timestamp': '2022-11-26T03:17:33.399Z', 'expiration': '2022-11-26T03:18:43.399Z', 'matcherFee': '0.11888377', 'signature': '3tavERJnoJgdngHYePwD3w9vXYmM6SQfikyK6Y8hQqg2ffT13bwQdeoJsYC2mXewQRv1tNMVBTcn5xj4ubSMiGRm', 'matcherFeeAssetId': 'Atqv59EYzjFGuitKVnMRk6H8FukjoV3ktPorbEys25on'}}}, 'timestamp': 1669432654263, 'datetime': '2022-11-26T03:17:34.263Z', 'symbol': 'ETH/BTC', 'id': 'Fnc9rbxg87Uik6dSZeLnDzoYfY9C7g1mRjBxcwwniTrT', 'order': '8BcAp5PDVAZup99Sdo4it8hTm93HjCeTrTkSCkQ4sAYq', 'type': None, 'side': 'sell', 'takerOrMaker': None, 'price': 0.0729996, 'amount': 0.06195336, 'cost': 0.004522570498656, 'fee': {'cost': 0.11888377, 'currency': 'WX'}, 'fees': [{'currency': 'WX', 'cost': 0.11888377}]}
total trades: 100
buy trades:   0
sell trades:  100
maker trades: 0
taker trades: 0
```

All trades returned by `fetch_trades` have 'side' as 'sell'  for all pairs I have tried. This is similar to issue #15794. All trades also have 'takerOrMaker' as `None`, but I don't know if this value could be defined or not.

By taking a look at 'Market Trades' on the Waves Exchange trading website (https://waves.exchange/trading/spot/ETH_BTC) one clearly sees that both buy and sell trades are occurring.

The probable reason for this problem is that `parse_trade` method has this code in lines 2014-2017 in wavesexchange.py:
```
        if self.safe_string(order1, 'senderPublicKey') == self.apiKey:
            order = order1
        else:
            order = order2
```
It makes no sense to make a comparison to the `self.apiKey`, because these are public market trades that can be obtained without defining any api keys (see https://api.wavesplatform.com/v0/transactions/exchange?amountAsset=474jTeYx2r2Va35794tCScAXWJG9hU2HcgxzMowaZUnu&priceAsset=8LQW8f7P5d5PZM7GtZEBgaqRPGSzS3DfPuiXrURJ4AJS&limit=100). The comparison probably makes much more sense for `fetch_my_trades` though.

Consequently 'order2' is almost always chosen as `order`, from which other information, including order id and 'side' are extracted. It looks like for 'order2' 'type' is always 'sell', leading to the problem in question. 

As a side note, defining a singular order id ('order' in the ccxt dictionary) for market trades probably does not make too much sense, because every public trade corresponds to two orders - maker and taker. Perhaps it would be better to include both if they are available like in this case?

In conclusion, `parse_trade` should probably not be doing the parsing work for both `fetch_trades` and `fetch_my_trades`. Whether 'side' of the trade can actually be determined from dictionary returned by the exchange, I do not unfortunately know, but it would be nice have it if possible.
